cpp/src/rolling/rolling.cu

```diff
@@ -70,10 +70,42 @@ process_rolling_window(column_device_view input,
   // for CUDA 10.0 and below (fixed in CUDA 10.1)
   volatile cudf::size_type count = 0;
 
-  for (size_type j = start_index; j < end_index; j++) {
-    if (op == aggregation::COUNT_ALL || !has_nulls || input.is_valid(j)) { count++; }
+  if (!has_nulls) {
+    count = end_index - start_index;
+  } else {
+    for (size_type j = start_index; j < end_index; j++) {
+      if (input.is_valid(j)) { count++; }
+    }
   }
 
+  bool output_is_valid                      = ((end_index - start_index) >= min_periods);
+  output.element<OutputType>(current_index) = count;
```

Comment:
Good catch. I shouldn't be setting this.

Reply:
In that case, I suggest returning early and looping only when the output is valid :)
It's possible now since we don't use the count to determine validity.

Reply:
Yep, that's a good one. Fixed.